### PR TITLE
docs: add docs for build —target

### LINKED ISSUE #33143: build --target does not have docs
feature added in https://github.com/moby/moby/pull/32496

### PR DESCRIPTION
**Pull request by tonistiigi**
fixes https://github.com/moby/moby/issues/33143

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

**Review by dnephin — APPROVED** [review]
LGTM

File: docs/reference/commandline/build.md [hunk 1]
```diff
@@ -454,6 +455,22 @@ more `--add-host` flags. This example adds a static address for a host named
 
     $ docker build --add-host=docker:10.180.0.1 .
 
+### Specifying target build stage (--target)
+
+When building a Dockerfile with multiple build stages, `--target` can be used to
+speficy an intermediate build stage by name as a final stage for the resulting
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: s/speficy/specify/

File: docs/reference/commandline/build.md [hunk 2]
```diff
@@ -454,6 +455,22 @@ more `--add-host` flags. This example adds a static address for a host named
...
+
+```Dockerfile
+FROM debian AS build-env
+...
+
+FROM alpine AS production-env
+...
+```
+
+    $ docker build -t mybuildimage --target build-env .
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: Can you put this in fences with a `bash` language-hint?

    ```bash
    $ docker build -t mybuildimage --target build-env .
    ```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left some comments.

This also needs to be added to the man-page; https://github.com/moby/moby/blob/master/man/docker-build.1.md

**Comment by tonistiigi** [comment]
@thaJeztah PTAL

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**